### ISSUE TITLE
LoRA fine-tuning orchestrator with progress tracking

### ISSUE THREAD

**Issue by boersmamarcel** [boersmamarcel/chatty2#186]
## Summary

Orchestrate LoRA fine-tuning of specialist models using candle-lora (pure Rust) with SFT and DPO datasets. This is the core training engine triggered by the "Recalibrate Models" button in the Train settings page. Manages sequential training of multiple specialists with progress reporting.

**Design decision**: Training uses `candle-lora` — pure Rust, no Python/Unsloth/Axolotl dependency. Adapter weights are saved as safetensors and loaded directly by Candle at inference time. No GGUF conversion needed.

### Training Orchestrator

- [ ] Create `TrainingOrchestrator` entity that coordinates the full recalibration pipeline:
  1. Build SFT datasets (via #184)
  2. Build DPO datasets (via #185)
  3. For each specialist with sufficient data: run LoRA fine-tuning via candle-lora
  4. Evaluate adapted models
  5. Hot-swap successful adapters
- [ ] Sequential training (one model at a time to stay within memory budget)
- [ ] Skip specialists with insufficient training data (configurable minimum, default: 50 SFT examples)
- [ ] Emit `TrainingOrchestratorEvent` for UI progress updates:
  ```rust
  enum TrainingOrchestratorEvent {
      PipelineStarted { total_specialists: usize },
      DatasetBuildProgress { phase: String, progress: f32 },
      TrainingStarted { specialist: SpecialistType, index: usize, total: usize },
      TrainingProgress { specialist: SpecialistType, epoch: usize, loss: f32, progress: f32 },
      TrainingCompleted { specialist: SpecialistType, metrics: TrainingMetrics },
      EvaluationStarted { specialist: SpecialistType },
      EvaluationCompleted { specialist: SpecialistType, score: f32 },
      PipelineCompleted { trained: usize, skipped: usize, failed: usize },
      PipelineError { error: String },
      PipelineCancelled,
  }
  ```

### candle-lora Training Loop

- [ ] Implement LoRA adapter initialization per Candle model architecture:
  - Use `candle-lora` crate to wrap attention layers with LoRA
  - Target: Q/K/V projection layers (standard LoRA targets)
- [ ] SFT training loop:
  - Forward pass through Candle model with LoRA layers
  - Cross-entropy loss on next-token prediction
  - AdamW optimizer (implemented in `candle-nn`)
  - Gradient accumulation for effective batch size
- [ ] DPO training loop (after SFT):
  - Reference model = base (frozen), policy model = LoRA-adapted
  - DPO loss: `−log σ(β * (log π_θ(y_w|x) − log π_ref(y_w|x) − log π_θ(y_l|x) + log π_ref(y_l|x)))`
  - Only LoRA weights are updated
- [ ] Training hyperparameters per specialist (configurable with sensible defaults):
  - LoRA rank: 8
  - LoRA alpha: 16
  - Learning rate: 2e-4 (SFT), 5e-5 (DPO)
  - Epochs: 3 (SFT), 1 (DPO)
  - Batch size: 4 (with gradient accumulation if needed)
  - β (DPO): 0.1

### Adapter Storage

- [ ] Save LoRA adapter weights as safetensors after training
- [ ] Storage path: `{data_dir}/adapters/{specialist_type}/v{version}/adapter.safetensors`
- [ ] Save training metadata alongside: hyperparams, loss curve, data stats, timestamp
- [ ] Checkpoint saving between epochs for crash recovery

### Memory & Compute Management

- [ ] Run training on Tokio blocking thread pool (`spawn_blocking`) to avoid blocking UI
- [ ] Only one model loaded for training at a time
- [ ] Unload model after training before loading next specialist
- [ ] Apple Silicon: use Metal backend via `candle-metal` for GPU-accelerated training
- [ ] Cancellation support via `Arc<AtomicBool>` (consistent with StreamManager pattern)

### Progress Tracking

- [ ] Per-model training progress (epoch/batch level)
- [ ] Overall pipeline progress (models completed / total)
- [ ] Loss curve tracking for diagnostics
- [ ] Estimated time remaining based on throughput
- [ ] Training log persistence for debugging

## Implementation Notes

- `candle-lora` wraps existing Candle layers — each architecture (#182) needs LoRA-compatible layer definitions
- DPO requires two forward passes per batch (reference + policy) — memory doubles during DPO
- For 1.5B-3B models with LoRA rank 8: ~2M trainable params, feasible on M4 even on CPU
- Metal backend strongly recommended for reasonable training times on Apple Silicon
- Phase 2 #166 (candle-lora training loop) is the direct prerequisite — this issue extends it to the orchestrator level

## References

- [candle-lora crate](https://github.com/EricLBuehler/candle-lora)
- [DPO Paper](https://arxiv.org/abs/2305.18290)
- [Candle (HuggingFace Rust ML)](https://github.com/huggingface/candle)

## Depends On

- #184 (SFT dataset builder)
- #185 (DPO dataset builder)
- #182 (specialist registry + Candle model architectures)
- Phase 2 #166 (candle-lora infrastructure)
- Phase 2 #163 (CandleLocalModel for inference during evaluation)
